mozilla-b2g/gaia
# Bug 881141 - hdpi/xhdpi resolution assets and layout for Usage app

-   To test this patch on HVGA / WVGA devices:
  1.  `$ make install-gaia APP=costcontrol GAIA_DEV_PIXELS_PER_PX=1.5`
  
  What `GAIA_DEV_PIXELS_PER_PX` do is to replace image files by @1.5x assets when exist.
  Images will be too large (and break the layout) if we missed to set background-size for them in the patch.
  If this patch works well, we will get correct layout on both HVGA and WVGA (and ideally qHD, but AFAIK we don't have to confirm it at this time) screen.
-   Testing on nightly is also possible:
  1.  `$ make DEBUG=1 GAIA_DEV_PIXELS_PER_PX=1.5`
  2.  enter `about:config` and set `layout.css.devPixelsPerPx` to `1.5`
  3.  set responsive design window to 320 x 533 px (which is CSS pixel dimension for 480 x 800 device pixel dimension).
  
  But, of course, for apps using specific APIs, not all features can be tested.
